### PR TITLE
doc: Created page on avoiding the `UserWarning` issue

### PR DESCRIPTION
**Pull request by mistShard**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [x ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #338 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.
I simply completed a draft of a page that explains the `UserWarning` issue that can come up when using groupbys in pandas and tips on how to avoid it.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
thanks @mistShard - you've still got changes from your other PR here, could you rebase please?

so:
```
git fetch upstream
git rebase -i upstream/main
```
then, in the interactive window, only keep the commits you want (i.e. 2aeab37), then
```
git push origin HEAD -f
```

File: docs/other/group_by.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Tips for Avoiding the `UserWarning`
+To ensure efficiency and avoid warnings similar to those seen in Approach 1, data scientists should follow these practices:
+
+1. Decompose Complex Operations: Break down complex transformations into simpler steps. In this case, keep the `.agg` method simple.  Compute new columns first, then use these columns in aggregation or other operations.
+2. Avoid Redundant Computations: If an operation (like addition) is used multiple times, compute it once and store the result in a new column.
+3. Leverage Built-in Functions: Use built-in functions provided by the DataFrame library. In this case, using the `with_columns()` method allows you to pre-compute before grouping and aggregation.
+
+By following these guidelines, you can are sure to avoid the error: `UserWarning: Found complex group-by expression, which can't be expressed efficiently with the pandas API. If you can, please rewrite your query such that group-by aggregations are simple (e.g. mean, std, min, max, ...)`.
+
+Happy grouping! 🫡
```
Comment (MarcoGorelli) [inline docs/other/group_by.md:95]: 😄 love this

**Comment by mistShard** [comment]
> thanks @mistShard - you've still got changes from your other PR here, could you rebase please?
> 
> so:
> 
> ```
> git fetch upstream
> git rebase -i upstream/main
> ```
> 
> then, in the interactive window, only keep the commits you want (i.e. [2aeab37](https://github.com/narwhals-dev/narwhals/commit/2aeab37d85192188baad7d1cd8e62d3ff7af173f)), then
> 
> ```
> git push origin HEAD -f
> ```

I tried doing that but I have run into some issues (Note that before I added, committed, and pushed the changes in this PR, I had deleted all the files and references to the docs associated with the prior PR). After running `git fetch upstream` and `git rebase -i upstream/main` I deleted references to the other commits from the file that came up in the interactive window and kept the present commit.

However, when I ran `git push origin HEAD -f` I got the error 


> thanks @mistShard - you've still got changes from your other PR here, could you rebase please?
> 
> so:
> 
> ```
> git fetch upstream
> git rebase -i upstream/main
> ```
> 
> then, in the interactive window, only keep the commits you want (i.e. [2aeab37](https://github.com/narwhals-dev/narwhals/commit/2aeab37d85192188baad7d1cd8e62d3ff7af173f)), then
> 
> ```
> git push origin HEAD -f
> ```

I believe I've been able to do that now

File: CUsershpDocumentsWorkOpenTeams [hunk 2]
```diff
@@ -0,0 +1,340 @@
+{
```
Comment (MarcoGorelli) [inline CUsershpDocumentsWorkOpenTeams:1]: not sure what this file is, i think you can remove it?

File: plot.ipynb [hunk 3]
```diff
@@ -0,0 +1,326 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Chart Visualization\n",
```
Comment (MarcoGorelli) [inline plot.ipynb:7]: this file too

Comment (mistShard) [reply plot.ipynb]: I've deleted both.

File: docs/other/group_by.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    pd_df = pd.DataFrame(data)
+
+    def approach_1(df_any):
+        df = nw.from_native(df_any)
+
+        # Pay attention to this next line using group_by()
+        df = df.group_by('a').agg((nw.col('b')+nw.col('c')).sum().alias('d'))
+        return nw.to_native(df)
+
+    print(approach_1(pd_df))
```
Comment (FBruzzesi) [inline docs/other/group_by.md:25]: Two tips in here which apply also for `approach_2`:

- We can use `@nw.narwhalify` and avoid `from|to_native` operation to display
- If we split the operation in multiple lines I think it is easier to see the difference between the two approaches

Putting it together

```suggestion
    @nw.narwhalify
    def approach_1(df):
        return (df
            .group_by('a')
            .agg(
                (nw.col('b')+nw.col('c')).sum().alias('d')
             )
        )
    print(approach_1(pd_df))
```

Comment (FBruzzesi) [inline docs/other/group_by.md:25]: Two tips in here which apply also for `approach_2`:

- We can use `@nw.narwhalify` and avoid `from|to_native` operation to display
- If we split the operation in multiple lines I think it is easier to see the difference between the two approaches

Putting it together

```suggestion
    @nw.narwhalify
    def approach_1(df):
        return (df
            .group_by('a')
            .agg(
                (nw.col('b')+nw.col('c')).sum().alias('d')
             )
        )
    print(approach_1(pd_df))
```

File: docs/other/group_by.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+Both Approaches shown above return the exact same result, but Approach 1 is inefficient and returns the following error message.
+
+```
+UserWarning: Found complex group-by expression, which can't be expressed efficiently with the pandas API. If you can, please rewrite
+           your query such that group-by aggregations are simple (e.g. mean, std, min, max, ...).
+```
+
+You might rightly ask, what makes the first approach inefficient and the second approach efficient? To understand why Approach 2 is more efficient than Approach 1, it's essential to analyze the operations and transformations applied to the DataFrame in *line 11* of each approach. Let's break it down:
+
+## Approach 1
```
Comment (FBruzzesi) [inline docs/other/group_by.md:55]: Should we call this something along the lines of _the polars' way_ approach?!

**Review by FBruzzesi — COMMENTED** [review]
I really like this tutorial! 👌

I left a couple of comments just as ideas. 

As a follow up we could add even more examples, e.g. in the TPCH queries I remember moving one operation _after_ the `group_by`:

```diff
- .group_by(nw.sum("a") **2)...
+ .group_by(nw.sum("a")).with_columns(nw.col("a")**2)
```

**Pull request closed without merging**